metrics-proxy/src/main/java/ai/vespa/metricsproxy/service/MetricsParser.java

```diff
@@ -142,25 +143,42 @@ static private void handleValue(JsonParser parser, Instant timestamp, Consumer c
 
     private static Map<DimensionId, String> parseDimensions(JsonParser parser,
                                                             Map<Long, Map<DimensionId, String>> uniqueDimensions) throws IOException {
-        List<Map.Entry<String, String>> dims = new ArrayList<>();
-        int keyHash = 0;
-        int valueHash = 0;
+
+        List<Dimension> dimensions = new ArrayList<>();
+
         for (parser.nextToken(); parser.getCurrentToken() != JsonToken.END_OBJECT; parser.nextToken()) {
             String fieldName = parser.getCurrentName();
             JsonToken token = parser.nextToken();
+
             if (token == JsonToken.VALUE_STRING){
                 String value = parser.getValueAsString();
-                dims.add(Map.entry(fieldName, value));
-                keyHash ^= fieldName.hashCode();
-                valueHash ^= value.hashCode();
+                dimensions.add(Dimension.of(fieldName, value));
             } else if (token == JsonToken.VALUE_NULL) {
                 // TODO Should log a warning if this happens
             } else {
                 throw new IllegalArgumentException("Dimension '" + fieldName + "' must be a string");
             }
         }
-        Long uniqueKey = (((long) keyHash) << 32) | (valueHash & 0xffffffffL);
-        return uniqueDimensions.computeIfAbsent(uniqueKey, key -> dims.stream().collect(Collectors.toUnmodifiableMap(e -> toDimensionId(e.getKey()), Map.Entry::getValue)));
+        return uniqueDimensions.computeIfAbsent(dimensionsHashCode(dimensions),
+                                                key -> dimensions.stream().collect(Collectors.toUnmodifiableMap(
+                                                        dim -> toDimensionId(dim.id), dim -> dim.value)));
+    }
+
+    static long dimensionsHashCode(List<Dimension> dimensions) {
+        int keyHash = 0;
+        int valueHash = 0;
+
+        for (Dimension dim : dimensions) {
+            keyHash += dim.id.hashCode();
+            valueHash += dim.value.hashCode();
```

Comment:
Very good catch @gjoranv. This was way to naive.
We can not treat key and value separately.
I suggest using com.yahoo.compress.Hasher.xxh3 and just keep one 64bit.
Either that or follow @hakonhall's suggestion.